src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Internal/MsQuicApi.cs

```diff
@@ -26,81 +32,22 @@ internal unsafe sealed class MsQuicApi
-
-            ListenerOpenDelegate =
-                new ListenerOpenDelegate(new DelegateHelper(vtable->ListenerOpen).ListenerOpen);
-            ListenerCloseDelegate =
-                Marshal.GetDelegateForFunctionPointer<ListenerCloseDelegate>(
-                    vtable->ListenerClose);
-            ListenerStartDelegate =
-                new ListenerStartDelegate(new DelegateHelper(vtable->ListenerStart).ListenerStart);
-            ListenerStopDelegate =
-                new ListenerStopDelegate(new DelegateHelper(vtable->ListenerStop).ListenerStop);
-
-            ConnectionOpenDelegate =
-                new ConnectionOpenDelegate(new DelegateHelper(vtable->ConnectionOpen).ConnectionOpen);
-            ConnectionCloseDelegate =
-                Marshal.GetDelegateForFunctionPointer<ConnectionCloseDelegate>(
-                    vtable->ConnectionClose);
-            ConnectionSetConfigurationDelegate =
-                new ConnectionSetConfigurationDelegate(new DelegateHelper(vtable->ConnectionSetConfiguration).ConnectionSetConfiguration);
-            ConnectionShutdownDelegate =
-                new ConnectionShutdownDelegate(new DelegateHelper(vtable->ConnectionShutdown).ConnectionShutdown);
-            ConnectionStartDelegate =
-                new ConnectionStartDelegate(new DelegateHelper(vtable->ConnectionStart).ConnectionStart);
-
-            StreamOpenDelegate =
-                new StreamOpenDelegate(new DelegateHelper(vtable->StreamOpen).StreamOpen);
-            StreamCloseDelegate =
-                Marshal.GetDelegateForFunctionPointer<StreamCloseDelegate>(
-                    vtable->StreamClose);
-            StreamStartDelegate =
-                new StreamStartDelegate(new DelegateHelper(vtable->StreamStart).StreamStart);
-            StreamShutdownDelegate =
-                new StreamShutdownDelegate(new DelegateHelper(vtable->StreamShutdown).StreamShutdown);
-            StreamSendDelegate =
-                new StreamSendDelegate(new DelegateHelper(vtable->StreamSend).StreamSend);
-            StreamReceiveCompleteDelegate =
-                new StreamReceiveCompleteDelegate(new DelegateHelper(vtable->StreamReceiveComplete).StreamReceiveComplete);
-            StreamReceiveSetEnabledDelegate =
-                new StreamReceiveSetEnabledDelegate(new DelegateHelper(vtable->StreamReceiveSetEnabled).StreamReceiveSetEnabled);
-
-            var cfg = new RegistrationConfig
+            ApiTable = apiTable;
+
+            fixed (byte* pAppName = s_appName)
             {
-                AppName = ".NET",
-                ExecutionProfile = QUIC_EXECUTION_PROFILE.QUIC_EXECUTION_PROFILE_LOW_LATENCY
-            };
+                var cfg = new QUIC_REGISTRATION_CONFIG {
+                    AppName = (sbyte*)pAppName,
+                    ExecutionProfile = QUIC_EXECUTION_PROFILE.QUIC_EXECUTION_PROFILE_LOW_LATENCY
```

Comment:
@ManickaP what are your thoughts on MsQuic possibly renaming all our enums to remove the type name from the value. i.e. `QUIC_EXECUTION_PROFILE.QUIC_EXECUTION_PROFILE_LOW_LATENCY` -> `QUIC_EXECUTION_PROFILE.LOW_LATENCY`?

Reply:
I'd love that, we actually had it like that before. And IMHO it would make the code much more readable since it's more in the spirit of C#.

Reply:
https://github.com/microsoft/msquic/pull/2650